src/legacy/event-loop-spinner.ts

```diff
@@ -0,0 +1,16 @@
+export class EventLoopSpinner {
+  private lastSpin: number;
+
+  constructor(private thresholdMs: number = 50) {
```

Comment:
Maybe 10? :)

Reply:
You think that blocking for 50ms is too much?

Reply:
I think 10-20 is the sweet spot based on this fun nonsense:
```js
async function trivial(x, y) {
  const f = x + y + x * y - Math.random();
}
async function nonTrivial(x, y) {
  try {
    throw new Error({x, y});
  } catch (e) {
    const v = e.message + '' + Math.random();
  }
  const f = x + y + x * y - Math.random();
}
async function run(fn, iterations) {
  while(iterations--) {
    await fn(Math.random(), Math.random());
  }
}
async function main() {
  let i = 50;
  while(i--) {
    const start = Date.now();
    const iterations = i * i * i;
    await run(trivial, iterations);
    console.log(`${iterations} absolutely trivial async iterations took ${Date.now() - start} ms`);
  }
  i = 50;
  while(i--) {
    const start = Date.now();
    const iterations = i * i * i;
    await run(nonTrivial, iterations);
    console.log(`${iterations} non-trivial async iterations took ${Date.now() - start} ms`);
  }
  return 'done';
}

main().then(console.log).catch(console.error);
```
```bash
...
64000 absolutely trivial async iterations took 20 ms
...
32768 absolutely trivial async iterations took 10 ms
...
3375 non-trivial async iterations took 20 ms
...
1728 non-trivial async iterations took 12 ms
```
But what does it all mean? ;) Let's go for `10`

Reply:
cool - will do 10